subprojects/core/src/main/java/org/gradle/execution/TaskPathProjectEvaluator.java

```diff
@@ -33,6 +33,11 @@ public void configure(ProjectInternal project) {
         if (cancellationToken.isCancellationRequested()) {
             throw new BuildCancelledException();
         }
+        // Need to configure intermediate parent projects for configure-on-demand
+        ProjectInternal parentProject = project.getParent();
+        if (parentProject != null && parentProject != project.getRootProject()) {
+            configure(project.getParent());
```

Comment:
💄 should reuse `parentProject` here

Reply:
Of course! You can tell it's been a while since I've written code. Fixed.